### PR TITLE
Reorganize subtotals for cart, checkout summary and order summary

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | reorganize subtotals for cart, checkout summary and order summary |
| Type? | improvement |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-723 |
| How to test? | Make an order |

### CODE REVIEW

File: src/Adapter/Cart/CartPresenter.php [hunk 1]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
-                'amount' => $this->priceFormatter->format(
-                    $total_including_tax - $total_excluding_tax
-                ),
-            ];
-        }
+        $subtotals['products'] = [
+            'type' => 'products',
+            'label' => $this->translator->trans('Products', [], 'Cart'),
+            'amount' => $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS),
+            'value' =>  $this->priceFormatter->format(($cart->getOrderTotal(true, Cart::ONLY_PRODUCTS)))
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: Missed the last coma ^^

File: src/Adapter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
-        if (Configuration::get('PS_TAX_DISPLAY')) {
-            $subtotals['tax'] = [
-                'type' => 'tax',
-                'label' => $this->translator->trans('Tax', [], 'Cart'),
-                'amount' => $this->priceFormatter->format(
-                    $total_including_tax - $total_excluding_tax
-                ),
-            ];
-        }
+        $subtotals['products'] = [
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: `array(`

File: src/Adapter/Cart/CartPresenter.php [hunk 3]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
             $subtotals['gift_wrapping'] = [
                 'type' => 'gift_wrapping',
                 'label' => $this->translator->trans('Gift wrapping', [], 'Cart'),
-                'amount' => $cart->getGiftWrappingPrice($this->includeTaxes()) != 0
-                                ? $this->priceFormatter->format($cart->getGiftWrappingPrice($this->includeTaxes()))
-                                : $this->translator->trans('Free', [], 'Cart'),
+                'amount' => $giftWrappingPrice,
+                'value' => ($giftWrappingPrice > 0)
+                    ? $this->priceFormatter->format($giftWrappingPrice)
+                    : $this->translator->trans('Free', [], 'Cart'),
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: 👍

File: src/Adapter/Cart/CartPresenter.php [hunk 4]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
+
         $shipping_cost = $cart->getTotalShippingCost(null, $this->includeTaxes());
         $subtotals['shipping'] = [
             'type' => 'shipping',
             'label' => $this->translator->trans('Shipping', [], 'Cart'),
-            'amount' => $shipping_cost != 0
+            'amount' => $shipping_cost,
+            'value' => $shipping_cost != 0
                             ? $this->priceFormatter->format($shipping_cost)
                             : $this->translator->trans('Free', [], 'Cart'),
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: Bad indent

File: src/Adapter/Cart/CartPresenter.php [hunk 5]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
 
         $products_count = array_reduce($products, function ($count, $product) {
             return $count + $product['quantity'];
         }, 0);
 
         $summary_string = $products_count === 1 ?
-                          $this->translator->trans('1 product', [], 'Cart') :
-                          sprintf($this->translator->trans('%d products', [], 'Cart'), $products_count)
+                          $this->translator->trans('1 item', [], 'Cart') :
+                          sprintf($this->translator->trans('%d items', [], 'Cart'), $products_count)
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: Bad indent

File: src/Adapter/Cart/CartPresenter.php [hunk 6]
```diff
@@ -215,71 +215,92 @@ public function present($cart)
...
         ;
 
+        // TODO: move it to a common parent, since it's copied in OrderPresenter
+        $labels = [
```
Comment (Shudrum) [inline src/Adapter/Cart/CartPresenter.php]: `array(`

File: src/Adapter/Order/OrderPresenter.php [hunk 7]
```diff
@@ -133,15 +132,15 @@ private function getAmounts(Order $order)
         $total_products = ($this->includeTaxes()) ? $order->total_products_wt : $order->total_products;
         $subtotals['products'] = [
             'type' => 'products',
-            'label' => $this->translator->trans('Products', [], 'Cart').' '.$tax_label,
+            'label' => $this->translator->trans('Products', [], 'Cart').' ',
```
Comment (Shudrum) [inline src/Adapter/Order/OrderPresenter.php]: We still need the last `.' '`?

File: src/Adapter/Order/OrderPresenter.php [hunk 8]
```diff
@@ -133,15 +132,15 @@ private function getAmounts(Order $order)
...
             'amount' => $total_products,
             'value' => $this->priceFormatter->format($total_products),
         ];
 
         $shipping_cost = ($this->includeTaxes()) ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
         $subtotals['shipping'] = [
             'type' => 'shipping',
-            'label' => $this->translator->trans('Shipping and handling', [], 'Cart').' '.$tax_label,
+            'label' => $this->translator->trans('Shipping and handling', [], 'Cart').' ',
```
Comment (Shudrum) [inline src/Adapter/Order/OrderPresenter.php]: We still need the last `.' '`?

File: src/Adapter/Order/OrderPresenter.php [hunk 9]
```diff
@@ -150,22 +149,23 @@ private function getAmounts(Order $order)
         if ((float)$discount_amount) {
             $subtotals['discounts'] = [
                 'type' => 'discount',
-                'label' => $this->translator->trans('Discount', [], 'Cart').' '.$tax_label,
+                'label' => $this->translator->trans('Discount', [], 'Cart').' ',
```
Comment (Shudrum) [inline src/Adapter/Order/OrderPresenter.php]: Repetitas

File: src/Adapter/Order/OrderPresenter.php [hunk 10]
```diff
@@ -150,22 +149,23 @@ private function getAmounts(Order $order)
...
                 'amount' => $discount_amount,
                 'value' => $this->priceFormatter->format($discount_amount),
             ];
         }
 
         $amounts['subtotals'] = $subtotals;
 
-        $amounts['total'] = [
+        $amounts['totals'] = array();
+        $amounts['totals']['total'] = [
```
Comment (Shudrum) [inline src/Adapter/Order/OrderPresenter.php]: `array(`

File: themes/classic/templates/checkout/order-confirmation.tpl [hunk 11]
```diff
@@ -17,90 +17,22 @@
...
-                            {/if}
-                            {hook h='displayProductPriceBlock' product=$product type="unit_price"}
-                        </td>
-                        <td>{$product.price}</td>
-                        <td>{$product.quantity}</td>
-                        <td>{$product.total}</td>
-                    </tr>
-                {/foreach}
-            </table>
+      <div class="row">
```
Comment (Shudrum) [inline themes/classic/templates/checkout/order-confirmation.tpl]: Where did this `<div>` is closed?

Comment (julienbourdeau) [inline themes/classic/templates/checkout/order-confirmation.tpl]: 😨

File: themes/classic/templates/checkout/cart.tpl [hunk 12]
```diff
@@ -3,6 +3,7 @@
 {block name='content'}
 
   <section id="main">
+    {$cart|dump}
```
Comment (Shine-neko) [inline themes/classic/templates/checkout/cart.tpl]: ?

Comment (julienbourdeau) [inline themes/classic/templates/checkout/cart.tpl]: 🚶 💨

File: themes/classic/templates/checkout/order-confirmation.tpl (hunk 11 above)
Comment (julienbourdeau) [inline themes/classic/templates/checkout/order-confirmation.tpl]: fixed here: https://github.com/PrestaShop/PrestaShop/pull/5550/files#diff-217a04faba9eb65ffb111829cdcd2bbaR35

File: src/Adapter/Cart/CartPresenter.php (hunk 4 above)
Comment (mickaelandrieu) [inline src/Adapter/Cart/CartPresenter.php]: also, `[]` should be `array()` you can use phcsfixer with `long_array_syntax` enabled :)

**Comment by mickaelandrieu** [comment]
@julienbourdeau for coding style stuff you can do

```
php bin/php-cs-fixer fix --level=symfony --fixers=long_array_syntax src/Adapter/Cart/CartPresenter.php
php bin/php-cs-fixer fix --level=symfony --fixers=long_array_syntax src/Adapter/Order/OrderPresenter.php
```

**Comment by Shudrum** [comment]
Conflicts :rage4: ^^

**Pull request merged**